app/src/main/kotlin/io/sweers/catchup/ui/fragments/SmmryFragment.kt

```diff
@@ -328,10 +266,10 @@ data class SmmryStorageEntry(
 interface SmmryDao {
 
   @Query("SELECT * FROM $TABLE WHERE url = :url")
-  fun getItem(url: String): Maybe<SmmryStorageEntry>
+  fun getItem(url: String): SmmryStorageEntry?
```

Comment:
I'm not sure what the best answer for daos should be, but I'm almost tempted to do things like:

    @Query("SELECT * FROM $TABLE WHERE url = :url")
    fun getItemBlocking(url: String): SmmryStorageEntry?

and then

    suspend fun SmmryDao.getItem(url: String) = withContext(Dispatchers.IO) { getItemBlocking(url) }

so that you're in suspending mode at the lowest layers of your own usage

Reply:
Neat! cd24c73